core/services/registrysyncer/syncer.go

```diff
@@ -41,6 +46,11 @@ type registrySyncer struct {
 	registryAddress string
 	peerWrapper     p2ptypes.PeerWrapper
 
+	orm ORM
+
+	updateChan     chan *LocalRegistry
+	testUpdateChan chan bool
```

Comment:
Not a huge fan of this. Could we remove it and instead mock the ORM in tests and verify if AddLocalRegistry() was called?